src/components/interface/filter/FilterMenu.jsx

```diff
@@ -0,0 +1,96 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import Button from '@material-ui/core/Button';
+import FilterIcon from '@material-ui/icons/PlaylistAdd';
+import Menu from '@material-ui/core/Menu';
+import MenuItem from '@material-ui/core/MenuItem';
+
+import SubMenu from './SubMenu';
+
+class FilterMenu extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      anchorEl: null
+    };
+  }
+
+  handleMenuOpen = (event) => {
+    this.setState({ anchorEl: event.currentTarget });
+  };
+
+  handleMenuClose = () => {
+    this.setState({ anchorEl: null });
+  };
+
+  renderChoices = (name, opt) => {
+    const { filtered } = this.props;
+    return opt.map(o => ({
+      checked: filtered[name].includes(o),
+      label: o,
+      name,
+      value: o
+    }));
+  };
+
+  renderMenuItems = options => Object.keys(options).map((name, index) => {
+    const { handleFilterChecked } = this.props;
+    const choices = this.renderChoices(name, options[name]);
+    const key = `${name}-${index}`;
+    return (
+      <SubMenu
+        caption={name}
+        choices={choices}
+        key={key}
+        onChange={handleFilterChecked} />
+    );
+  })
+
+  render() {
+    const { anchorEl } = this.state;
+    const { handleFilterClearAll, options } = this.props;
+    const anchorOrigin = { horizontal: 'left', vertical: 'bottom' };
+    const transformOrigin = { horizontal: 'left', vertical: 'top' };
+    const menuItems = options ? this.renderMenuItems(options) : [];
```

Comment:
So I think this could just be `const menuItems = options && this.renderMenuItems`.

React does not render false or undefined : https://react-cn.github.io/react/tips/false-in-jsx.html. so there's not really much fo a need for this default array to pop up.